packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/route.ts

```diff
@@ -0,0 +1,44 @@
+import { ApiKeyType, MedusaError } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+import { refetchApiKey } from "../../helpers"
+import { LinkMethodRequest } from "@medusajs/types/src"
+import { linkSalesChannelsToApiKeyWorkflow } from "@medusajs/core-flows"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<LinkMethodRequest>,
+  res: MedusaResponse
+) => {
+  const { add, remove } = req.validatedBody
+  const apiKey = await refetchApiKey(req.params.id, req.scope, ["id", "type"])
+
+  if (apiKey.type !== ApiKeyType.PUBLISHABLE) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      "Sales channels can only be associated with publishable API keys"
+    )
+  }
```

Comment:
q: shouldn't this be part of the workflow?

Reply:
@riqwan I just kept the existing logic, but I agree, it might be better if it's part of the workflow